content/data/scripts/ui_topics.lua

```diff
@@ -21,24 +22,47 @@ local function weaponStats(weaponClass)
 	local baseDamage = weaponClass.Damage
 	if weaponClass.OuterRadius > 0 then
 		-- This weapon has a shockwave, which gives it additional damage on a direct hit.
-		-- TODO: Most weapons have shockwave damage equal to their base damage, but not all.
-		local bonusDamage = baseDamage
+		-- Added formula to calculate shockwave damage -- WW
+		if weaponClass.ShockwaveDamage and weaponClass.ShockwaveDamage > 0 then
+			bonusDamage = weaponClass.ShockwaveDamage
+		elseif weaponClass.ShockwaveDamage == 0 or weaponClass.ShockwaveDamage == nil then
+			bonusDamage = baseDamage
+		end
 		baseDamage = baseDamage + bonusDamage
 	end
+
 	local velocity = weaponClass.Speed
-	local range = velocity * weaponClass.LifeMax
-	local rof = math.floor(100 / weaponClass.FireWait) / 100
+	local rof = utils.round(1 / weaponClass.FireWait, 2)
+	local range = math.min(weaponClass.Range, (velocity * weaponClass.LifeMax))
+
+	-- new code to calculate volley size -- WW
+	local isSwarmer, swarmcount, swarmwait = weaponClass:getSwarmInfo()
+	local isCorkscrew, cscrewcount, cscrewwait, cscrewrotate, cscrewtwist = weaponClass:getCorkscrewInfo()
+	local swarmproxy = math.max(swarmcount, 1) --swarmcount returns -1 if invalid
+
+	local cscrewproxy = 1
+	if isCorkscrew then
+		cscrewproxy = cscrewcount
+	end
```

Comment:
Instead of having proxy variables we can do something like

```
if not isCorkscrew then
    cscrewcount = 1
end
````

Can do the same with swarmproxy

`swarmcount = math.max(swarmcount, 1)`

Reply:
Done.